src/core/main.c

```diff
@@ -2064,6 +2061,9 @@ int main(int argc, char *argv[]) {
                 arg_serialization = safe_fclose(arg_serialization);
                 fds = fdset_free(fds);
 
+                /* Reopen the console */
+                (void) make_console_stdio();
```

Comment:
We are executing some unknown binary here, so it's reasonable to reset as much state as possible. In particular it might be using stdout/stderr to communicate.